ansible_collections/arista/avd/roles/eos_designs/templates/base/local-users.j2

```diff
@@ -10,4 +10,7 @@ local_users:
 {%     elif local_users[local_user].no_password is defined and local_users[local_user].no_password == True %}
     no_password: {{ local_users[local_user].no_password }}
 {%      endif %}
+{%     if local_users[local_user].ssh_key is defined %}
+    ssh_key: {{ local_users[local_user].ssh_key }}
+{%      endif %}
```

Comment:
The ```endif``` indentation is not aligning with the ```if```. Also use ```arista.avd.defined``` instead of ```defined```.

Reply:
Doh. I copied the role chunk and think about it not being right. I'll fix both of them.

Reply:
Yeah, the old template has wrong indentation. Don't fix the defined test in the old template though. But for ssh_key use arista.avd.defined

Reply:
You don't want me to clean up the whole file while I'm touching it?
```
--- a/ansible_collections/arista/avd/roles/eos_designs/templates/base/local-users.j2
+++ b/ansible_collections/arista/avd/roles/eos_designs/templates/base/local-users.j2
@@ -2,15 +2,15 @@ local_users:
 {% for local_user in local_users | arista.avd.natural_sort %}
   {{ local_user }}:
     privilege: {{ local_users[local_user].privilege }}
-{%     if local_users[local_user].role is defined %}
+{%     if local_users[local_user].role is arista.avd.defined %}
     role: {{ local_users[local_user].role }}
-{%      endif %}
-{%    if local_users[local_user].sha512_password is defined %}
+{%     endif %}
+{%     if local_users[local_user].sha512_password is arista.avd.defined %}
     sha512_password: {{ local_users[local_user].sha512_password }}
-{%     elif local_users[local_user].no_password is defined and local_users[local_user].no_password == True %}
+{%     elif local_users[local_user].no_password is arista.avd.defined and local_users[local_user].no_password is arista.avd.defined(true) %}
     no_password: {{ local_users[local_user].no_password }}
-{%      endif %}
-{%     if local_users[local_user].ssh_key is defined %}
+{%     endif %}
+{%     if local_users[local_user].ssh_key is arista.avd.defined %}
     ssh_key: {{ local_users[local_user].ssh_key }}
-{%      endif %}
+{%     endif %}
 {% endfor %}
```

Reply:
Actually, some knobs use defined because of the logic. So @ClausHolbechArista or @carlbuchmann should be able to comment on this. I am not sure about this refactor. You can correct the indentation with out any problem but when it comes to ```defined``` filter, I am not sure if we can change them to ```arista.avd.defined``` for all knobs in this template. For ssh_key, please use arista.avd.defined.

Reply:
I have pushed a new patch fixing all the whitespace, and only my if condition.